server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -4484,9 +4484,30 @@ public Durability getTranslogDurability() {
     // we can not protect with a lock since we "release" on a different thread
     private final AtomicBoolean flushOrRollRunning = new AtomicBoolean();
 
+    /**
+     * Checks if the shard need to be Refreshed depending on Translog constraints.
+     * Each Translog type can have it's own decider
+     * @return {@code true} if the shard should be Refreshed
+     */
+    public boolean shouldRefreshShard() {
+        final Engine engine = getEngineOrNull();
+        if (engine != null) {
+            try {
+                return engine.translogManager().shouldRefreshShard();
+            } catch (final AlreadyClosedException e) {
+                // we are already closed, no need to Refresh
+            }
+        }
+        return false;
+    }
```

Comment:
method semantics looks wrong to me. It's swallowing and exception while always returning false

Reply:
Yes, I was not sure if we want to raise the exception here, since this check is called after every write.